mithril-signer/src/configuration.rs

```diff
@@ -76,34 +77,39 @@ pub struct Configuration {
 
 impl Configuration {
     /// Return the CardanoNetwork value from the configuration.
-    pub fn get_network(&self) -> Result<CardanoNetwork, ConfigError> {
+    pub fn get_network(&self) -> StdResult<CardanoNetwork> {
         CardanoNetwork::from_code(self.network.clone(), self.network_magic)
-            .map_err(|e| ConfigError::Message(e.to_string()))
+            .with_context(|| "Could not read Network from configuration.")
     }
 
     /// Create the SQL store directory if not exist and return the path of the
     /// SQLite3 file.
-    pub fn get_sqlite_file(&self) -> PathBuf {
+    pub fn get_sqlite_file(&self) -> StdResult<PathBuf> {
         let store_dir = &self.data_stores_directory;
 
         if !store_dir.exists() {
-            std::fs::create_dir_all(store_dir).unwrap();
+            std::fs::create_dir_all(store_dir).with_context(|| {
+                format!(
+                    "Could not create directory '{}' for Sqlite3 file.",
+                    store_dir.display()
+                )
+            })?;
         }
 
-        self.data_stores_directory.join(SQLITE_FILE)
+        Ok(self.data_stores_directory.join(SQLITE_FILE))
     }
 
     /// Create era reader adapter from configuration settings.
     pub fn build_era_reader_adapter(
         &self,
         chain_observer: Arc<dyn ChainObserver>,
     ) -> StdResult<Arc<dyn EraReaderAdapter>> {
-        Ok(EraReaderAdapterBuilder::new(
+        EraReaderAdapterBuilder::new(
             &self.era_reader_adapter_type,
             &self.era_reader_adapter_params,
         )
         .build(chain_observer)
-        .map_err(|e| ConfigError::Message(format!("build era adapter failed {e}")))?)
+        .with_context(|| "Configuration: can not create era adapter.".to_string())
```

Comment:
The adapter type and params would be useful in the context message

Reply:
added `Display` for `EraAdapterType`.